launcher/src/test/java/engineering/everest/lhotse/functionaltests/scenarios/ApplicationFunctionalTests.java

```diff
@@ -58,25 +62,41 @@ void adminWillBeProvisioned() {
         usersRepository.findByUsernameIgnoreCase(adminUserName).orElseThrow();
     }
 
+    @Test
+    void metricsEndpointPublishesAxonMetrics() {
+        apiRestTestClient.createAdminUserAndLogin();
+
+        webTestClient.get().uri("/actuator/metrics/commandBus.successCounter")
+                .header("Authorization", "Bearer " + apiRestTestClient.getAccessToken())
+                .exchange()
+                .expectStatus().isEqualTo(OK);
+    }
+
     @Test
     void organizationsAndUsersCanBeCreated() {
         apiRestTestClient.createAdminUserAndLogin();
         var newOrganizationRequest = new NewOrganizationRequest("ACME", "123 King St", "Melbourne",
                 "Vic", "Oz", "3000", null, null, null, null);
         var newUserRequest = new NewUserRequest("user@example.com", "password", "Captain Fancypants");
 
-        var organizationId = apiRestTestClient.createOrganization(newOrganizationRequest, CREATED);
+        var organizationId = apiRestTestClient.createRegisteredOrganization(newOrganizationRequest, CREATED);
         var userId = apiRestTestClient.createUser(organizationId, newUserRequest, CREATED);
-        apiRestTestClient.getUser(organizationId, userId, OK);
+        apiRestTestClient.getUser(userId, OK);
     }
 
     @Test
-    void metricsEndpointPublishesAxonMetrics() {
-        apiRestTestClient.createAdminUserAndLogin();
+    void newUsersCanRegisterTheirOrganizationAndCreateNewUsersInTheirOrganization() throws InterruptedException {
+        apiRestTestClient.logout();
+        var registerOrganizationRequest = new RegisterOrganizationRequest("Alice's Art Artefactory", "123 Any Street", "Melbourne", "Victoria", "Australia", "3000", "http://alicesartartefactory.com", "Alice", "+61 422 123 456", "alice@example.com", "alicerocks");
+        var organizationRegistrationResponse =  apiRestTestClient.registerNewOrganization(registerOrganizationRequest, CREATED);
+        var newOrganizationId = organizationRegistrationResponse.getNewOrganizationId();
+        var pendingRegistration = pendingRegistrationsRepository.findByOrganizationId(newOrganizationId); // Confirmation code is "emailed", but we can cheat and pull it from the DB.
 
-        webTestClient.get().uri("/actuator/metrics/commandBus.successCounter")
-                .header("Authorization", "Bearer " + apiRestTestClient.getAccessToken())
-                .exchange()
-                .expectStatus().isEqualTo(OK);
+        apiRestTestClient.confirmOrganizationRegistration(newOrganizationId, pendingRegistration.getConfirmationCode(), OK);
+
+        apiRestTestClient.login("alice@example.com", "alicerocks");
+        Thread.sleep(15000L); // Allow event processing to catch up. CI is _ridiculously_ slow.
+        var newUserRequest = new NewUserRequest("bob@example.com", "bobalsorocks", "My name is Bob");
+        apiRestTestClient.createUser(newOrganizationId, newUserRequest, CREATED);
```

Comment:
I was thinking the same, that Travis might have isolation issues.

I'll try and get the logs tomorrow.

Reply:
Travis CI builds are definitely interfering with each other. The build just passed because one of the VMs was queued longer for creation.

Trying to get some log output to prove it.... multicast is probably enabled in their Google Cloud infrastructure.

Reply:
I've confirmed that the builds are isolated. However, there are two issues:
 - the saga is processed asynchronously, so the test does need to sleep for at least half a second for the subsequent events to complete processing
 - there seems to be a transaction boundary issue with the projection even handlers. This is what I'm focusing on now.

Reply:
I cannot reproduced with my 9 year old laptop. Did you fix it?

Reply:
Nope, it's still reproducible. I haven't made much progress - too many other priorities.